database/src/main/kotlin/no/nav/su/se/bakover/database/revurdering/RevurderingPostgresRepo.kt

```diff
@@ -1,33 +1,50 @@
-package no.nav.su.se.bakover.database
+package no.nav.su.se.bakover.database.revurdering
 
 import com.fasterxml.jackson.module.kotlin.readValue
 import kotliquery.Row
+import no.nav.su.se.bakover.common.UUID30
 import no.nav.su.se.bakover.common.objectMapper
 import no.nav.su.se.bakover.common.periode.Periode
+import no.nav.su.se.bakover.database.EksterneIverksettingsstegEtterUtbetalingMapper
+import no.nav.su.se.bakover.database.Session
 import no.nav.su.se.bakover.database.beregning.PersistertBeregning
+import no.nav.su.se.bakover.database.hent
+import no.nav.su.se.bakover.database.hentListe
+import no.nav.su.se.bakover.database.oppdatering
 import no.nav.su.se.bakover.database.søknadsbehandling.SøknadsbehandlingRepo
+import no.nav.su.se.bakover.database.tidspunkt
+import no.nav.su.se.bakover.database.uuid
+import no.nav.su.se.bakover.database.withSession
+import no.nav.su.se.bakover.domain.NavIdentBruker
 import no.nav.su.se.bakover.domain.NavIdentBruker.Saksbehandler
-import no.nav.su.se.bakover.domain.behandling.BeregnetRevurdering
-import no.nav.su.se.bakover.domain.behandling.OpprettetRevurdering
-import no.nav.su.se.bakover.domain.behandling.Revurdering
-import no.nav.su.se.bakover.domain.behandling.RevurderingTilAttestering
-import no.nav.su.se.bakover.domain.behandling.SimulertRevurdering
+import no.nav.su.se.bakover.domain.brev.BrevbestillingId
+import no.nav.su.se.bakover.domain.journal.JournalpostId
 import no.nav.su.se.bakover.domain.oppdrag.simulering.Simulering
 import no.nav.su.se.bakover.domain.oppgave.OppgaveId
+import no.nav.su.se.bakover.domain.revurdering.BeregnetRevurdering
+import no.nav.su.se.bakover.domain.revurdering.IverksattRevurdering
+import no.nav.su.se.bakover.domain.revurdering.OpprettetRevurdering
+import no.nav.su.se.bakover.domain.revurdering.Revurdering
+import no.nav.su.se.bakover.domain.revurdering.RevurderingTilAttestering
+import no.nav.su.se.bakover.domain.revurdering.SimulertRevurdering
 import no.nav.su.se.bakover.domain.søknadsbehandling.Søknadsbehandling
 import java.util.UUID
 import javax.sql.DataSource
 
 interface RevurderingRepo {
     fun hent(id: UUID): Revurdering?
+    fun hentRevurderingForBehandling(behandlingId: UUID): Revurdering?
+    fun hentRevurderingForUtbetaling(utbetalingId: UUID30): IverksattRevurdering?
     fun lagre(revurdering: Revurdering)
 }
 
 enum class RevurderingsType {
     OPPRETTET,
-    BEREGNET,
+    BEREGNET_INNVILGET,
+    BEREGNET_AVSLAG,
     SIMULERT,
     TIL_ATTESTERING,
+    IVERKSATT,
```

Comment:
TIL_ATTESTERING og IVERKSATT er ikke delt opp i INNVILGET og AVSLAG? Er de kun INNVILGET forslår jeg å spesifisere det nå.

Reply:
Jeg skjønner det sånn at vi ikke attesterer ny behandlingsinformasjon, men kun vedtak. Men alt dette er i en tidlig fase og vil bli gjennomgått på nytt. Så vi lar det være slik det er for nå.